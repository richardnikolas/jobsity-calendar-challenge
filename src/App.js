import React from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { Switch, Route, Redirect } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import { ThemeProvider } from '@material-ui/styles';
import configureStore, { history } from './store';
import { routes } from './shared/constants/routes';
import { muiTheme, palette } from './shared/styles/theme';
import Calendar from './features/calendar/views/Calendar';

const useStyles = makeStyles({
  root: {
    position: 'relative',
    overflowX: 'hidden',
    height: '100%',
    backgroundColor: palette.background.main,
  }
});

const store = configureStore();

function App() {
  const classes = useStyles();

  return (
    <Provider store={store}>
      <ThemeProvider theme={muiTheme}>
        <div className={classes.root} id={'rootApp'}>
          <ConnectedRouter history={history}>          
            <Route path="*">
              <Switch>
                <Route 
                  exact 
                  path={routes.home.path} 
                  render={Calendar} />
                <Redirect path="*" to={routes.home.path} />
              </Switch>
            </Route>          
          </ConnectedRouter>
        </div>
      </ThemeProvider>
    </Provider>
  )
}

export default App;
