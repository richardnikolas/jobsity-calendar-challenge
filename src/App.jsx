import React from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { Switch, Route, Redirect } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import { ThemeProvider } from '@material-ui/styles';
import configureStore, { history } from './store';
import { routes } from './shared/constants/routes';
import { muiTheme, palette } from './shared/styles/theme';
import withContext from './shared/hoc/withContext';
import HomePage from './features/calendar/views/HomePage';

const useStyles = makeStyles({
  root: {
    position: 'relative',
    overflowX: 'hidden',
    height: '100%',
    backgroundColor: palette.background.main,
  }
});

const store = configureStore();

function App() {
  const classes = useStyles();

  return (
    <Provider store={store}>
      <ThemeProvider theme={muiTheme}>
        <div className={classes.root} id={'rootApp'}>
          <ConnectedRouter history={history}>          
            <Route path="*">
              <Switch>
                <Route
                  exact 
                  path={routes.home.path}
                  render={withContext(HomePage)} />
                <Redirect path="*" to={routes.home.path} />
              </Switch>
            </Route>          
          </ConnectedRouter>
        </div>
      </ThemeProvider>
    </Provider>
  )
}

export default App;
