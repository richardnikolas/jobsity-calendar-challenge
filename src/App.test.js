import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import App from './App';

describe('Render <App />', () => {
  let container = null;
  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('Should render <App /> snapshot', () => {
    act(() => {
      render(<App />, container);
    });

    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});