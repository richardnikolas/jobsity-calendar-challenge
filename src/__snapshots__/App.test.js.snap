// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <App /> Should render <App /> snapshot 1`] = `
"<div class=\\"makeStyles-root-1\\" id=\\"rootApp\\">
  <div class=\\"container\\">
    <section class=\\"header-section\\">
      <div class=\\"header\\">
        <div class=\\"greeting-container\\">
          <h1 class=\\"greeting\\">Good afternoon</h1>
        </div>
      </div>
    </section>
    <section class=\\"sidebar-section\\">
      <div class=\\"sidebar\\"><button class=\\"btn-new-reminder\\">
          <div class=\\"btn-info\\"><i class=\\"btn-icon\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\">
                <path d=\\"M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.49 2 2 6.49 2 12s4.49 10 10 10 10-4.49 10-10S17.51 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z\\"></path>
              </svg></i>
            <p>New Reminder</p>
          </div>
        </button></div>
    </section>
    <section class=\\"calendar-section\\">
      <div class=\\"makeStyles-container-2\\">
        <div class=\\"makeStyles-child-3\\">Sunday</div>
        <div class=\\"makeStyles-child-3\\">Monday</div>
        <div class=\\"makeStyles-child-3\\">Tuesday</div>
        <div class=\\"makeStyles-child-3\\">Wednesday</div>
        <div class=\\"makeStyles-child-3\\">Thursday</div>
        <div class=\\"makeStyles-child-3\\">Friday</div>
        <div class=\\"makeStyles-child-3\\">Saturday</div>
      </div>
      <div class=\\"calendar\\">
        <div class=\\"calendar-day inactiveDays\\">
          <p>28</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day inactiveDays\\">
          <p>29</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day inactiveDays\\">
          <p>30</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>1</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>2</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>3</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>4</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>5</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>6</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>7</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>8</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>9</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>10</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>11</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>12</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>13</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>14</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>15</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>16</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>17</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>18</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>19</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>20</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>21</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>22</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>23</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>24</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>25</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>26</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>27</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>28</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>29</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>30</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day\\">
          <p>31</p>
          <div class=\\"reminders-area\\"></div>
        </div>
        <div class=\\"calendar-day inactiveDays\\">
          <p>1</p>
          <div class=\\"reminders-area\\"></div>
        </div>
      </div>
    </section>
  </div>
</div>"
`;
